packages/experiments/src/components/Microfeedback/Microfeedback.base.tsx

```diff
@@ -0,0 +1,161 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { withSlots } from '../../Foundation';
+import { Stack } from '../../Stack';
+import { classNamesFunction } from '../../Utilities';
+import { IconButton, IButtonStyles, DefaultButton } from 'office-ui-fabric-react/lib/Button';
+import { IStackStyles } from 'office-ui-fabric-react/lib/Stack';
+import { Callout } from 'office-ui-fabric-react/lib/Callout';
+import { FocusZone, FocusZoneDirection } from 'office-ui-fabric-react/lib/FocusZone';
+import { List } from 'office-ui-fabric-react/lib/List';
+import { Text, ITextStyles } from 'office-ui-fabric-react/lib/Text';
+import {
+  IMicrofeedbackProps,
+  IMicrofeedbackViewProps,
+  IMicrofeedbackStyleProps,
+  IMicrofeedbackStyles,
+  VoteType
+} from './Microfeedback.types';
+import { initializeIcons } from '@uifabric/icons';
+
+const getClassNames = classNamesFunction<IMicrofeedbackStyleProps, IMicrofeedbackStyles>();
+
+const microfeedbackStyles: IStackStyles = {
+  root: [
+    {
+      margin: 8,
+      float: 'right'
+    }
+  ]
+};
+
+const microfeedbackItemStyles: IButtonStyles = {
+  root: [
+    {
+      border: '0px',
+      width: '100%'
+    }
+  ]
+};
+
+const microfeedbackTextStyles: ITextStyles = {
+  root: [
+    {
+      margin: '10px'
+    }
+  ]
+};
+
+export interface IMicrofeedbackState extends IMicrofeedbackViewProps {
+  vote: VoteType;
+  isFollowupVisible: boolean;
+}
+
+export class MicrofeedbackBase extends React.Component<IMicrofeedbackProps, IMicrofeedbackState> {
+  // ref's will be linked to each of the icons for callout placement
+  private dislikeRef = React.createRef<HTMLDivElement>();
+  private likeRef = React.createRef<HTMLDivElement>();
+
+  constructor(props: IMicrofeedbackProps) {
+    super(props);
+
+    initializeIcons();
+
+    this.state = {
+      // initial state of icons is neutral and followup is not visible
+      vote: 'no_vote',
+      isFollowupVisible: false
+    };
+  }
+
+  public render() {
+    const likeIcon = this.state.vote === 'like' ? 'LikeSolid' : 'Like';
+    const dislikeIcon = this.state.vote === 'dislike' ? 'DislikeSolid' : 'Dislike';
+    const hideThumbsDownCallout = this.state.vote !== 'dislike' || !this.state.isFollowupVisible;
+    const hideThumbsUpCallout = this.state.vote !== 'like' || !this.state.isFollowupVisible;
+
+    const classNames = getClassNames(this.props.styles, {
+      theme: this.props.theme
+    });
+
+    const likeVote = () => {
```

Comment:
These could also be class functions like `_vote` (`_likeVote`, `_dislikeVote`) so that new functions aren't created every time `render` is called.

Reply:
Fixed